nifi-nar-bundles/nifi-azure-bundle/nifi-azure-services-api/src/main/java/org/apache/nifi/services/azure/data/explorer/KustoIngestionResult.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.services.azure.data.explorer;
+
+public enum KustoIngestionResult {
+    SUCCEEDED("Succeeded"),PARTIALLY_SUCCEEDED("PartiallySucceeded"),FAILED("Failed"),DEFAULT("Default");
+    private String status;
+
+    KustoIngestionResult(String status) {
+        this.status = status;
+    }
```

Comment:
General question, since this can be used in streaming and batch mode ,

what happens when we stream data and poll for results ? Does data get "blocked" in this case. 

In spark, in queued mode for example we do ingestFromBlob and do not poll for ingestion.

Reply:
This is an interesting problem. During my testing with streaming mode since I was testing with a relatively small subset of data didn't experience the "data block" as such. My initial thoughts during streaming ingestion was to skip polling for ingestion result but the fall back to queued mode would have been an escape incase of failures and hence thought of adding the polling for ingestion result for managed streaming cases as well.

Reply:
@tanmaya-panda1 . There is a challenge!

Lets assume the user provides a lot of data streaming (e.g. 500MB every minute), polling for results may block as we have to ack.

As I understand , we have an option in NiFi to run realtime or batch.

Would it make sense to have a boolean parameter that says block polling for results ? Default this to false, which means it will keep going without looking for status.

User can set it to true and poll for results ? This way it is flexible. 

In the doc lets mention pros and cons of both

a) results will be polled and hence there may be a backlog if you poll
b) If you do not poll, then you may have errors (mitigate this through alerts that run looking for ingestion failures and notifying users ? )